src/runtime/node/dns/index.ts

```diff
@@ -0,0 +1,75 @@
+import noop from "../../mock/noop";
+import mock from "../../mock/proxy";
+import {
+  notImplemented,
+  notImplementedAsync,
+} from "src/runtime/_internal/utils";
+import type dns from "node:dns";
+import * as constants from "./constants";
+export * from "./constants";
+
+export const Resolver: typeof dns.Resolver =
+  mock.__createMock__("dns.Resolver");
+export const getDefaultResultOrder: typeof dns.getDefaultResultOrder = () =>
+  "verbatim";
+export const getServers: typeof dns.getServers = () => [];
+export const lookup: typeof dns.lookup = notImplementedAsync("dns.lookup");
+export const lookupService: typeof dns.lookupService =
+  notImplementedAsync("dns.lookupService");
+export const promises: typeof dns.promises =
+  mock.__createMock__("dns.promises");
```

Comment:
Most of non promisified mocks throw error it seems little bit inconsistent behavior if promises namespace is only (non throwing) behavior.

Maybe we can create `dns/promises` leveraging `notImplementedAsync` util?